### PR TITLE
Improvement create XWikiRights class

### PR DESCRIPTION
**Pull request by ebeutler**
https://synjira.atlassian.net/browse/CELDEV-380

### CODE REVIEW

File: src/main/java/com/celements/model/classes/fields/BooleanField.java [hunk 1]
```diff
@@ -12,11 +12,13 @@
 public final class BooleanField extends AbstractClassField<Boolean> implements
     CustomClassField<Boolean> {
 
+  private final String displayFormType;
```
Comment (msladek) [inline src/main/java/com/celements/model/classes/fields/BooleanField.java:15]: add new field to BooleanFieldTest

File: src/main/java/com/celements/web/classcollections/oldcore/IOldCoreClassDef.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.celements.web.classcollections.oldcore;
+
+import org.xwiki.component.annotation.ComponentRole;
+
+import com.celements.model.classes.ClassDefinition;
+
+@ComponentRole
+public interface IOldCoreClassDef extends ClassDefinition {
```
Comment (msladek) [inline src/main/java/com/celements/web/classcollections/oldcore/IOldCoreClassDef.java]: add javadoc, XWiki classes should never be added to any ClassPackage

**Review by msladek — CHANGES_REQUESTED** [review]

File: src/main/java/com/celements/web/classcollections/oldcore/IOldCoreClassDef.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package com.celements.web.classcollections.oldcore;
```
Comment (msladek) [inline src/main/java/com/celements/web/classcollections/oldcore/IOldCoreClassDef.java]: up until now we have used the package name "classes" for the new class definitions. perhaps rename the package to **com.celements.web.classes**

**Review by msladek — APPROVED** [review]

**Review by fpichler — APPROVED** [review]

File: src/main/java/com/celements/web/classes/oldcore/XWikiRightsClass.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  public static final String CLASS_DEF_HINT = CLASS_FN;
+
+  public static final String FIELD_NAME_ALLOW = "allow";
+  public static final String FIELD_PRETTY_NAME_ALLOW = "Allow/Deny";
+  public static final String FIELD_NAME_ACCESSLVL = "levels";
+  public static final String FIELD_PRETTY_NAME_ACCESSLVL = "Levels";
+  public static final String FIELD_NAME_USERS = "users";
+  public static final String FIELD_PRETTY_NAME_USERS = "Users";
+  public static final String FIELD_NAME_GROUPS = "groups";
+  public static final String FIELD_PRETTY_NAME_GROUPS = "Groups";
```
Comment (msladek) [inline src/main/java/com/celements/web/classes/oldcore/XWikiRightsClass.java]: using Strings for fields should be discouraged, better remove them. if really needed one could always use FIELD_X.getName()

**Review by msladek — CHANGES_REQUESTED** [review]

**Review by msladek — APPROVED** [review]

**Pull request merged**